db/db_impl/db_impl_open.cc

```diff
@@ -1002,12 +1002,17 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
 
       // We create a new batch and initialize with a valid prot_info_ to store
       // the data checksums
-      WriteBatch batch(0, 0, 8, 0);
+      WriteBatch batch;
 
       status = WriteBatchInternal::SetContents(&batch, record);
       if (!status.ok()) {
         return status;
       }
+      status = WriteBatchInternal::SetProtectionBytesPerKey(
+          &batch, 8 /* bytes_per_key */);
+      if (!status.ok()) {
+        return status;
+      }
```

Comment:
I felt this change more clearly expresses what was already happening: the checksum calculation happens on the copy of the `record` data. Without it, one might have guessed `SetContents()` calculates checksum directly on the `record` passed to it.

It is related to this PR because I needed to introduce `SetProtectionBytesPerKey()` anyways for use in `Write()`.